Update to the docs and examples (mainly c++ stuff)

Myles1 opened the pull request:
Fix a few typos/grammar things.
Add a comment to the docs
Add an example for compilation and running a c++ extension to the docs
Add c++ code for an example extension to the example directory

scoder commented:
Looks like the generated `rect.cpp` was accidentally committed.

scoder commented on docs/examples/tutorial/rectangle_cpp/rect.pyx:
```diff
@@ -0,0 +1,56 @@
+# disutils: language = c++
```
It's spelled `distutils`, not `disutils`.

scoder commented on docs/examples/tutorial/rectangle_cpp/setup.py:
```diff
@@ -0,0 +1,9 @@
+
+setup(ext_modules = cythonize(Extension(
+           "rect",                                # the extension name
+           sources=["rect.pyx", "Rectangle.cpp"], # the Cython source and
```
This is redundant because the configuration already exists in the source file. I think it's better to keep it there, so that this simplifies to `cythonize("rect.pyx")`.

scoder commented:
Very nice. Happy to merge it once the issues above are resolved.

scoder commented:
Oh, would also be cool to have a test that builds and executes this example, to make sure it keeps working. Not sure if there is a good precedent, but the `EmbedTest` class in `runtests.py` probably comes closest.

Myles1 commented:
I fixed each issue, and added a test for setup / usage / teardown. I'm not familiar enough with the cython tests codebase to be confident integrating it there. Any guidance for the test would be greatly appreciated.

Myles1 commented:
@scoder

gabrieldemarmiesse commented:
Hello! A pull request (#2296) is in progress to enable automatic testing of all the files in the `Cython/docs/examples` directory. Once it is merged, you will be able to drop your files in there and they will (hopefully if nothing goes wrong) be tested automatically. Thanks for your patience!
